client/app/scripts/components/topology-options.js

```diff
@@ -73,9 +73,26 @@ class TopologyOptions extends React.Component {
   renderOption(option) {
     const { activeOptions, currentTopologyId } = this.props;
     const optionId = option.get('id');
-    const activeValue = activeOptions && activeOptions.has(optionId)
-      ? activeOptions.get(optionId)
-      : option.get('defaultValue');
+
+    // Make the active value be the intersection of the available options
+    // and the active selection and use the default value if there is no
+    // overlap. It seems intuitive that active selection would always be a
+    // subset of available option, but the exception can happen when going
+    // back in time (making available options change, while not touching
+    // the selection).
+    // TODO: This logic should probably be made consistent with how topology
+    // selection is handled when time travelling, especially when the name-
+    // spaces are brought under category selection.
+    let activeValue = option.get('defaultValue');
+    if (activeOptions && activeOptions.has(optionId)) {
+      const activeSelection = makeSet(activeOptions.get(optionId));
+      const availableOptions = makeSet(option.get('options').map(o => o.get('value')));
+      const intersection = activeSelection.intersect(availableOptions);
+      if (!intersection.isEmpty()) {
+        activeValue = intersection.toJS();
+      }
+    }
```

Comment:
Actually I don't think it's worth the effort at this point (it would have to look a bit different as a global selector, working with all the options and not per option like here), but I'll add a _TODO_ for the future!